homeassistant/components/zwave_mqtt/entity.py

```diff
@@ -0,0 +1,272 @@
+            "Specific_command_class=%s, "
+            "Command_class=%s, Index=%s, Value type=%s, "
+            "Genre=%s as %s",
+            self._node.node_id,
+            self._node.node_generic,
+            self._node.node_specific,
+            self.primary.command_class,
+            self.primary.index,
+            self.primary.type,
+            self.primary.genre,
+            component,
+        )
+        self._entity_created = True
+
+        if component in PLATFORMS:
+            async_dispatcher_send(self._hass, f"zwave_new_{component}", self)
+
+    @property
+    def values_id(self):
+        """Identification for this values collection."""
+        return f"{self.primary.node.id}-{self.primary.value_id_key}"
+
+
+class ZWaveDeviceEntity(Entity):
+    """Generic Entity Class for a Z-Wave Device."""
+
+    def __init__(self, values):
+        """Initialize a generic Z-Wave device entity."""
+        self.values = values
+        self.options = values.options
+
+    @callback
+    def on_value_update(self):
+        """Call when a value is added/updated in the entity EntityValues Collection.
+
+        To be overridden by platforms needing this event.
+        """
+
+    async def async_added_to_hass(self):
+        """Call when entity is added."""
+        # add dispatcher and OZW listeners callbacks,
+        self.options.listen(EVENT_VALUE_CHANGED, self._value_changed)
+        self.options.listen(EVENT_INSTANCE_STATUS_CHANGED, self._instance_updated)
+        # add to on_remove so they will be cleaned up on entity removal
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, const.SIGNAL_DELETE_ENTITY, self._delete_callback
+            )
+        )
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, f"{self.values.values_id}_value_added", self._value_added
+            )
+        )
+
+    @property
+    def device_info(self):
+        """Return device information for the device registry."""
+        node = self.values.primary.node
+        node_instance = self.values.primary.instance
+        dev_id = create_device_id(node, self.values.primary.instance)
+        device_info = {
+            "identifiers": {(DOMAIN, dev_id)},
+            "name": create_device_name(node),
+            "manufacturer": node.node_manufacturer_name,
+            "model": node.node_product_name,
+        }
+        # device with multiple instances is split up into virtual devices for each instance
+        if node_instance > 1:
+            parent_dev_id = create_device_id(node)
+            device_info["name"] += f" - Instance {node_instance}"
+            device_info["via_device"] = (DOMAIN, parent_dev_id)
+        return device_info
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return {const.ATTR_NODE_ID: self.values.primary.node.node_id}
```

Comment:
Is this useful? We already have the device info page show this info.

Reply:
If the user needs it for automations, otherwise we can remove it, I think.

Reply:
No at this time the node id is not visible anywhere so this is helpfull untill the controlpanel is ready. No way to find which node id belongs to which device otherwise,

Reply:
Let's keep this for now.